runtime/src/accounts_background_service.rs

```diff
@@ -169,23 +171,64 @@ impl SnapshotRequestHandler {
                     shrink_time.stop();
                 }
 
+                let block_height = snapshot_root_bank.block_height();
+                let snapshot_type = if block_height
+                    % self.snapshot_config.full_snapshot_archive_interval_slots
+                    == 0
+                {
+                    *last_full_snapshot_slot = Some(snapshot_root_bank.slot());
+                    Some(SnapshotType::FullSnapshot)
+                } else if block_height
+                    % self
+                        .snapshot_config
+                        .incremental_snapshot_archive_interval_slots
+                    == 0
+                    && last_full_snapshot_slot.is_some()
+                {
+                    Some(SnapshotType::IncrementalSnapshot(
+                        last_full_snapshot_slot.unwrap(),
+                    ))
+                } else {
+                    None
+                };
+
                 // Generate an accounts package
+                let mut attempts = 0;
                 let mut snapshot_time = Measure::start("snapshot_time");
-                let r = snapshot_utils::snapshot_bank(
-                    &snapshot_root_bank,
-                    status_cache_slot_deltas,
-                    &self.accounts_package_sender,
-                    &self.snapshot_config.bank_snapshots_dir,
-                    &self.snapshot_config.snapshot_archives_dir,
-                    self.snapshot_config.snapshot_version,
-                    &self.snapshot_config.archive_format,
-                    hash_for_testing,
-                );
-                if r.is_err() {
+                loop {
+                    attempts += 1;
+                    let result = snapshot_utils::snapshot_bank(
+                        &snapshot_root_bank,
+                        status_cache_slot_deltas.clone(),
```

Comment:
Yep! I'm looking to avoid that. See this thread for previous discussion:
https://github.com/solana-labs/solana/pull/19401#discussion_r695107247

But now I am going to do a different implementation _without_ the retry loop, so I'll be able to remove the `.clone()` here.

Reply:
And now retry is gone, so `.clone()` is gone. heh.